enterprise/frontend/src/embedding-sdk/cli/snippets/express-server-snippet.ts

```diff
@@ -0,0 +1,147 @@
+import { HARDCODED_USERS } from "embedding-sdk/cli/constants/hardcoded-users";
+
+import {
+  HARDCODED_JWT_SHARED_SECRET,
+  USER_ATTRIBUTE_CUSTOMER_ID,
+} from "../constants/config";
+
+interface Options {
+  instanceUrl: string;
+  tenantIds: (number | string)[];
+}
+
+const DEFAULT_EXPRESS_SERVER_PORT = 4477;
+
+export const getExpressServerSnippet = (options: Options) => {
+  const users = HARDCODED_USERS.map((user, i) => ({
+    ...user,
+
+    // Assign one of the tenant id in the user's database to their Metabase user attributes.
+    // This is hard-coded for demonstration purposes.
+    ...(options.tenantIds[i] && {
+      [USER_ATTRIBUTE_CUSTOMER_ID]: options.tenantIds[i],
+    }),
+  }));
+
+  return `
+const express = require('express')
+const jwt = require('jsonwebtoken')
+const cors = require('cors')
+const session = require('express-session')
+
+const PORT = process.env.PORT || ${DEFAULT_EXPRESS_SERVER_PORT}
+
+const METABASE_INSTANCE_URL = '${options.instanceUrl}'
+
+const METABASE_JWT_SHARED_SECRET =
+  '${HARDCODED_JWT_SHARED_SECRET}'
+
+const USERS = ${JSON.stringify(users, null, 2)}
+
+const getUser = (email) => USERS.find((user) => user.email === email)
+
+const signUserToken = (user) =>
+  jwt.sign(
+    {
+      email: user.email,
+      first_name: user.firstName,
+      last_name: user.lastName,
+      groups: user.groups,
+      customer_id: user.customerId,
+      exp: Math.round(Date.now() / 1000) + 60 * 0.25,
```

Comment:
On the readme we have
`      exp: Math.round(Date.now() / 1000) + 60 * 10, // 10 minutes expiration`
I'm guessing here it means it's 15 seconds?
Any reason for the difference?

Reply:
@npretto no, this is taken directly from https://github.com/metabase/sso-examples/blob/master/nodejs-jwt-example/index.js (our SSO examples)